### DISCUSSION TITLE
Container will sometimes not start but recomposing will fix it

### DISCUSSION THREAD

**Discussion by FaykoB** [haugene/docker-transmission-openvpn#2379] [Q&A]
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [X] I have tried using dev branch

### Docker run config used

  transmission-openvpn:
    image: haugene/transmission-openvpn     
    container_name: mediaRequests_VPN
    devices:
      - /dev/net/tun:/dev/net/tun      
    cap_add:
      - NET_ADMIN
      - SYS_MODULE
      - ALL
    networks:
      - traefik_ProxyNetwork
      - backEnd_Network      
    ports:
      - '9091:9091'     
    volumes:
      - $DOCKERDIR/qbt/data:/config
      - /etc/localtime:/etc/localtime:ro
      - $DATADIR/Games:/games:rw
      - $DATADIR/Movies:/movies:rw
      - $DATADIR/Music:/music:rw
      - $DATADIR/Pictures:/pictures:rw
      - $DATADIR/Purgatory:/Download      
      - $DATADIR/TV:/tv:rw
      - $DATADIR/Anime:/anime:rw
    environment:
      - OPENVPN_PROVIDER=MULLVAD
      - OPENVPN_CONFIG=us_chi
      - OPENVPN_USERNAME=$OPENVPN_USERNAME
      - OPENVPN_PASSWORD=m
      - LOCAL_NETWORK=$LOCAL_NETWORK
      - TRANSMISSION_WEB_UI=flood-for-transmission
    logging:
      driver: json-file
      options:
        max-size: 10m
    sysctls:
      - net.ipv4.conf.all.src_valid_mark=1
      - net.ipv6.conf.all.disable_ipv6=0        
    labels:
      traefik.enable: true
      traefik.http.routers.transmission-rtr.tls: true
      traefik.http.routers.transmission.entryPoints: https
      traefik.http.routers.transmission.middlewares: auth@file
      traefik.http.routers.transmission.rule: Host(`transmission.$DOMAINNAME`)
      traefik.http.routers.transmission-rtr.service: api@internal
      traefik.http.services.transmission-svc.loadbalancer.server.port: 9091
      traefik.http.services.transmission-svc.loadbalancer.server.scheme: http  

### Current Behavior

my container starts and stops randomly:
Starting container with revision: 84941a9ea4663d8b2e1af3db1d50fe4f7fa8736e
Creating TUN device /dev/net/tun
mknod: /dev/net/tun: File exists
Using OpenVPN provider: MULLVAD
Running with VPN_CONFIG_SOURCE auto
No bundled config script found for MULLVAD. Defaulting to external config
Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.crfFG56zzX
Extracting configs to /tmp/tmp.pW5w3jK79w
Found configs for MULLVAD in /tmp/tmp.pW5w3jK79w/vpn-configs-contrib-main/openvpn/mullvad, will replace current content in /etc/openvpn/mullvad
Cleanup: deleting /tmp/tmp.crfFG56zzX and /tmp/tmp.pW5w3jK79w
Starting OpenVPN using config us_chi.ovpn
Modifying /etc/openvpn/mullvad/us_chi.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Setting OpenVPN credentials...
adding route to local network 192.168.0.1/24 via 172.20.0.1 dev eth0
RTNETLINK answers: Invalid argument
adding route to local network 192.168.20.1/24 via 172.20.0.1 dev eth0
RTNETLINK answers: Invalid argument
adding route to local network 192.168.30.1/24 via 172.20.0.1 dev eth0
RTNETLINK answers: Invalid argument
Mon Oct 31 21:24:58 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Mar 22 2022
Mon Oct 31 21:24:58 2022 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10
Mon Oct 31 21:24:58 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Mon Oct 31 21:24:58 2022 RESOLVE: Cannot resolve host address: us-chi-101.mullvad.net:1197 (Name or service not known)
Mon Oct 31 21:24:58 2022 RESOLVE: Cannot resolve host address: us-chi-101.mullvad.net:1197 (Name or service not known)
Mon Oct 31 21:25:03 2022 RESOLVE: Cannot resolve host address: us-chi-101.mullvad.net:1197 (Name or service not known)
Mon Oct 31 21:25:08 2022 RESOLVE: Cannot resolve host address: us-chi-101.mullvad.net:1197 (Name or service not known)
Mon Oct 31 21:25:08 2022 Could not determine IPv4/IPv6 protocol
Mon Oct 31 21:25:08 2022 SIGTERM[soft,init_instance] received, process exiting

### Expected Behavior

to start up normally with no connection issues. 

### How have you tried to solve the problem?

rebuilding or swapping to glueton/qbt

### Log output

Starting container with revision: 84941a9ea4663d8b2e1af3db1d50fe4f7fa8736e
Creating TUN device /dev/net/tun
mknod: /dev/net/tun: File exists
Using OpenVPN provider: MULLVAD
Running with VPN_CONFIG_SOURCE auto
No bundled config script found for MULLVAD. Defaulting to external config
Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.crfFG56zzX
Extracting configs to /tmp/tmp.pW5w3jK79w
Found configs for MULLVAD in /tmp/tmp.pW5w3jK79w/vpn-configs-contrib-main/openvpn/mullvad, will replace current content in /etc/openvpn/mullvad
Cleanup: deleting /tmp/tmp.crfFG56zzX and /tmp/tmp.pW5w3jK79w
Starting OpenVPN using config us_chi.ovpn
Modifying /etc/openvpn/mullvad/us_chi.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Setting OpenVPN credentials...
adding route to local network 192.168.0.1/24 via 172.20.0.1 dev eth0
RTNETLINK answers: Invalid argument
adding route to local network 192.168.20.1/24 via 172.20.0.1 dev eth0
RTNETLINK answers: Invalid argument
adding route to local network 192.168.30.1/24 via 172.20.0.1 dev eth0
RTNETLINK answers: Invalid argument
Mon Oct 31 21:24:58 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Mar 22 2022
Mon Oct 31 21:24:58 2022 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10
Mon Oct 31 21:24:58 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Mon Oct 31 21:24:58 2022 RESOLVE: Cannot resolve host address: us-chi-101.mullvad.net:1197 (Name or service not known)
Mon Oct 31 21:24:58 2022 RESOLVE: Cannot resolve host address: us-chi-101.mullvad.net:1197 (Name or service not known)
Mon Oct 31 21:25:03 2022 RESOLVE: Cannot resolve host address: us-chi-101.mullvad.net:1197 (Name or service not known)
Mon Oct 31 21:25:08 2022 RESOLVE: Cannot resolve host address: us-chi-101.mullvad.net:1197 (Name or service not known)
Mon Oct 31 21:25:08 2022 Could not determine IPv4/IPv6 protocol
Mon Oct 31 21:25:08 2022 SIGTERM[soft,init_instance] received, process exiting

### HW/SW Environment

```markdown
- OS: Synology DSM
- Docker: 3.9
```

### Anything else?

I'm using traefik as well but doesn't seem like i have issues connecting to these hosts but it just fails. Running compose a few times just resolves it.

**Comment by pkishino** [comment]
I can see it trying to create invalid routes above and not able to resolve the url.. this is synology related and dsm causes all kinds of strange stuff

**Reply by FaykoB** [reply]
sigh i feel like I fucked up by going the syno route. It doing weird / abnormal stuff compared to other distros has been the bane of my existence. 

Should I just keep it on force restart  and ignore the issue?

**Reply by FaykoB** [reply]
also i've had to mount /data + /config or it will create it's own temporary directory. It looks like both of these are for settings? Is that right or is there a way for me to combine it to one folder?

**Reply by pkishino** [reply]
/data is for transmission downloads, /config is the container config and settings.. both are configurable, please see documentation and/or the scripts
